spinalcordtoolbox/image.py

```diff
@@ -259,12 +292,21 @@ def __init__(self, param=None, hdr=None, orientation=None, absolutepath=None, di
         else:
             raise TypeError('Image constructor takes at least one argument.')
 
-        # TODO: In the future, we might want to check qform_code and enforce its value. Related to #2454
-        # Check qform_code
-        # if not self.hdr['qform_code'] in [0, 1]:
-        #     # Set to 0 (unknown)
-        #     self.hdr.set_qform(self.hdr.get_qform(), code=0)
-        #     self.header.set_qform(self.hdr.get_qform(), code=0)
+        # Make sure sform and qform are the same.
+        # Context: https://github.com/neuropoly/spinalcordtoolbox/issues/2429
+        if check_sform and not check_affines_match(self):
+            if self.absolutepath is None:
+                logger.error("Internal code has produced an image with inconsistent qform and sform "
+                             "please report this on github at https://github.com/neuropoly/spinalcordtoolbox/issues "
+                             " or on the SCT forum https://forum.spinalcordmri.org/.")
+            else:
```

Comment:
The logging error here looks possibly redundant with the `try`/`except` block inside `check_affines_match`. (See below)